### DISCUSSION TITLE
how to poll a custom function when the captive portal is active

### DISCUSSION THREAD

**Discussion by dncex** [Hieromon/AutoConnect#372] [How can I?]
In my application I do start the captive portal manually:
portal.begin();

This function is blocking, so I was wondering if there's a way to call a custom function while the captive portal is active (in my case it is a fuction to keep a serial communication active)

thanks in advance!

**Comment by Bighoneypot** [comment]
Why do you need the serial?

**Reply by dncex** [reply]
To keep the main board of the system informed about the ESP board status

**Comment by Hieromon** [comment]
Is this the callback feature you need?
Refs: [Sketch execution during the captive portal loop](https://hieromon.github.io/AutoConnect/adcpcontrol.html#sketch-execution-during-the-captive-portal-loop)
Here's a sketch that uses it to achieve a serial keepalive (Keep in mind that it is pseudo code):

```cpp
AutoConnect portal;

bool whileCP(void) {
  KeepAlive_to_Serial();
  return true;
}

void setup() {
  ...
  portal.whileCaptivePortal(whileCP);
  portal.begin();
  ...
}

void loop() {
  KeepAlive_to_Serial();
  portal.handleClient();
}
```

**Reply by dncex** [reply]
many thanks... it is exactly what i was looking for! It works perfectly!
